py/sentry_streams/pipeline.py

```diff
@@ -124,3 +125,13 @@ class Map(WithInput):
     # the symbol is just not present in the current code base.
     function: Union[MapFunction, str]
     step_type: StepType = StepType.MAP
+
+
+@dataclass
+class Filter(WithInput):
+    """
+    A simple Filter, taking a single input and either returning it or None as output
+    """
+
+    function: Callable[..., Any]
```

Comment:
I think this is not the right type. 
1. You still need to support the string version
2. The typed versionshould be a callable that takes a single argument of type Any and returns a bool: `Callable[[Any], bool]` Otherwise the type checker will not complain against any function type provided here.